src/pricer.py

```diff
@@ -371,11 +371,58 @@ def calc_rho(self, method: GreekMethod) -> float:
         raise NotImplementedError('Greeks are not implemented for tree method yet.')
 
 
-# todo: to be implemented
 class GenericPDEPricer(Pricer):
+    def __init__(self, contract: EuropeanContract, model: MarketModel, params: PDEParams):
+        self._derivative_type = contract.get_type()
+        self._bsPDE = BlackScholesPDE(model, params)
+        self.grid = self._bsPDE.grid
+        self._initial_spot = model.get_initial_spot()
+        self._strike = contract.get_strike()
+        self._interest_rate = model.get_rate()
+        self.t_step = params.t_step
+        self.und_step = params.s_step
+        self.S_min = params.S_min
+        self.S_max = params.S_max
+        self.ns_steps = int(np.round((self.S_max - self.S_min) / float(self.und_step)))  # Number of stock price steps
+        self.nt_steps = int(np.round(params.exp / float(self.t_step)))  # Number of time steps
+        self.method = params.method
+        self.setup_boundary_conditions = self._bsPDE.setup_boundary_conditions()
+
     def calc_fair_value(self) -> float:
-        raise NotImplementedError('Fair value is not implemented yet for GenericPDEPricer.')
 
+        if self.method.upper() == BSPDEMethod.EXPLICIT:
+            self._bsPDE.explicit_method()
+        elif self.method.upper() == BSPDEMethod.IMPLICIT:
+            self._bsPDE.implicit_method()
+        elif self.method.upper() == BSPDEMethod.CRANKNICOLSON:
+            self._bsPDE.crank_nicolson_method()
+        else:
+            raise ValueError("Invalid method. Use 'explicit', 'implicit', or 'crank_nicolson'.")
+
+        # linear interpolation
+        down = int(np.floor((self._initial_spot - self.S_min)/self.und_step))
+        up = int(np.ceil((self._initial_spot - self.S_min)/self.und_step))
+
+        if down == up:
+            return self.grid[1, down+1]
+        else:
+            return self.grid[1,  down+1] + (self.grid[1, up+1] - self.grid[1, down+1]) * \
+                   (self._initial_spot - self.S_min - down*self.und_step)/self.und_step
+
+    def calc_delta(self, method: GreekMethod) -> float:
```

Comment:
For now, you can simply remove greek methods (later add them back). The generic bump & reval way of calculating greeks have been implemented in the base Pricer class, so it will be inherited for PDE as well.

Reply:
this has been removed and the pricer has been renamed